Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/self_intersections.h

```diff
@@ -314,6 +402,24 @@ self_intersections_impl(const FaceRange& face_range,
 
   if(throw_on_SI)
     CGAL::box_self_intersection_d<CGAL::Sequential_tag>(box_ptr.begin(), box_ptr.end(), throwing_filter, cutoff);
+  else if(do_limit)
+  {
```

Comment:
in sequential mode a user could already write:
```
  std::size_t nbi=0;
  auto max_inter_counter = [&nbi, max_nb_reported](const std::pair<face_descriptor, face_descriptor>&)
  {
    if (++nbi == max_nb_reported)
      throw CGAL::internal::Throw_at_output_exception();
  };

  try{
    PMP::self_intersections<CGAL::Sequential_tag>(tm, boost::make_function_output_iterator(max_inter_counter));
  }
  catch (const CGAL::internal::Throw_at_output_exception&)
  {}
```
I'm surprised we require so many lines.